NeoML/include/NeoML/Dnn/DnnDistributed.h

```diff
@@ -77,10 +80,10 @@ class NEOML_API CDistributedTraining {
 	void Train();
 	// Returns last loss of `layerName` for all models
 	// `layerName` should correspond to CLossLayer, CCtcLossLayer or CCrfLossLayer
-	void GetLastLoss( const CString& layerName, CArray<float>& losses );
+	void GetLastLoss( const CString& layerName, CArray<float>& losses ) const;
 	// Returns last blobs of `layerName` for all models
 	// `layerName` should correspond to CSinkLayer
-	void GetLastBlob( const CString& layerName, CObjectArray<CDnnBlob>& blobs );
+	void GetLastBlob( const CString& layerName, CObjectArray<CDnnBlob>& blobs ) const;
```

Comment:
Shouldn't it be `CObjectArray<const CDnnBlob>& blobs` to ensure that no changes of internal state are possible through the const method? Btw, are new blobs allocated after Train() call or the same being reused? If the second option, modifying them from outside could at least lead to user data loss.

Reply:
The first, in a `sinkLayer` the last blobs would created every `runOnce` new.
It should not be any affect

Reply:
No a such problem
`// NOTE: Main thread waits while all tasks are done`